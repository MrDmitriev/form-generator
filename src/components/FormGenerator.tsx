import { useState, MouseEvent, ChangeEvent } from "react"
import { FORM_CONFIG_TAB, FORM_RESULT_TAB, TAB_LIST } from "../constants/tabs";
import FormConfig from "./FormConfig";
import FormResult from "./FormResult";
import Tab from "./Tab";



export default function FormGenerator() {
	const [activeTab, setActiveTab] = useState(FORM_CONFIG_TAB);
	const [formConfig, setFormConfid] = useState('');

	const handleTabClick = (e: MouseEvent<HTMLButtonElement>) => {
		const tabName = e.currentTarget.id;
		setActiveTab(tabName);
		console.log('tab name', tabName);
	}

	const handleConfigChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
		const value = e.currentTarget.value;
		setFormConfid(value);
		console.log('cnahge value', value);
	}
	return (
		<div className='form-generator'>
			<div className="tab">
				{TAB_LIST.map(tab => <Tab key={tab} tab={tab} isActive={tab === activeTab} onTabClick={handleTabClick} /> )}
			</div>

			{activeTab === FORM_CONFIG_TAB && <FormConfig onConfigChange={handleConfigChange} value={formConfig} />}
			{activeTab === FORM_RESULT_TAB && <FormResult formConfig={formConfig} />}

		</div>
	)
}
