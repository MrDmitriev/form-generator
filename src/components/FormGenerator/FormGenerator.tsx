import { useState, MouseEvent, ChangeEvent, useCallback } from "react"
import { FORM_CONFIG_TAB, FORM_RESULT_TAB, TAB_LIST } from "../../constants/tabs";
import FormConfig from "../FormConfig/FormConfig";
import FormResult from "../FormResult/FormResult";
import Tab from "./components/Tab";

/*
	Main component, which resolves all functionality related to generating forms.
	This component renders main skeleton: two tabs (config tab: where you can input form configuration
	and result tab: where you can see a generated form) and tab content. 
*/
export default function FormGenerator() {
	const [activeTab, setActiveTab] = useState(FORM_CONFIG_TAB);
	const [formConfig, setFormConfig] = useState('');

	const handleTabClick = useCallback(
		(e: MouseEvent<HTMLButtonElement>) => {
			const tabName = e.currentTarget.id;
			setActiveTab(tabName);
		},
		[]
	)

	const handleConfigChange = useCallback(
		(e: ChangeEvent<HTMLTextAreaElement>) => {
			const value = e.currentTarget.value;
			setFormConfig(value);
		},
		[]
	)

	const handleConfigSubmit = useCallback(
		() => {
			setActiveTab(FORM_RESULT_TAB);
		},
		[]
	)

	return (
		<div className='form-generator'>
			<div className="tab">
				{TAB_LIST.map(tab => <Tab key={tab} tab={tab} isActive={tab === activeTab} onTabClick={handleTabClick} /> )}
			</div>

			{activeTab === FORM_CONFIG_TAB
				&& <FormConfig
					onConfigChange={handleConfigChange}
					onConfigSubmit={handleConfigSubmit}
					value={formConfig}
				/>}
				
			{activeTab === FORM_RESULT_TAB && <FormResult formConfig={formConfig} />}

		</div>
	)
}
