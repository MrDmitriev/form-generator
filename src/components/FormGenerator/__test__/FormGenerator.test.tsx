import 'jsdom-global/register';
import renderer from 'react-test-renderer';
import FormGenerator from '../FormGenerator';

describe("FormGenerator component", () => {
  it(`should match snapshot`, () => {
	
		const wrapper = renderer.create(<FormGenerator />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});