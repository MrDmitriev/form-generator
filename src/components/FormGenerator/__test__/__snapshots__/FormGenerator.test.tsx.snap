// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGenerator component should match snapshot 1`] = `
<div
  className="form-generator"
>
  <div
    className="tab"
  >
    <button
      className="tablinks active"
      id="config"
      onClick={[Function]}
    >
      config
    </button>
    <button
      className="tablinks false"
      id="result"
      onClick={[Function]}
    >
      result
    </button>
  </div>
  <div
    className="tabcontent active"
    id="config"
  >
    <form
      className="form-configurator"
    >
      <textarea
        className="form-configurator__input"
        onChange={[Function]}
        placeholder="{'items': [], 'buttons': []}"
        value=""
      />
    </form>
    <button
      className="form-configurator__submit"
      onClick={[Function]}
    >
       Apply
    </button>
  </div>
</div>
`;
